internal/plugin/connectors/http/generic/connector.go

```diff
@@ -39,11 +41,22 @@ func (c *Connector) Connect(
 	}
 
 	// Add configured params to request
-	params, err := renderTemplates(c.config.QueryParams, credentialsByID)
+	queryParams, err := renderTemplates(c.config.QueryParams, credentialsByID)
 	if err != nil {
 		return fmt.Errorf("failed to render query params: %s", err)
 	}
-	r.URL.RawQuery = appendQueryParams(*r.URL, params)
+	r.URL.RawQuery = appendQueryParams(*r.URL, queryParams)
+
+	// Add OAuth1 Authorization Header to the request
+	oauth1params, err := renderTemplates(c.config.OAuth1, credentialsByID)
+	if err != nil {
+		return fmt.Errorf("failed to render OAuth1 params: %s", err)
+	}
+	oauthHeader, err := oauth.CreateOAuth1Header(oauth1params, r)
+	if err != nil {
+		return fmt.Errorf("failed to create OAuth1 'Authorization' header: %s", err)
+	}
```

Comment:
It almost seems like that `renderTemplates` and `CreateOAuth1Header` should be in a separate smaller function